hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -1329,12 +1327,12 @@ public ListKeysResult listKeys(String volumeName, String bucketName,
       while (currentCount < maxKeys + 1 && keyIter.hasNext()) {
         kv = keyIter.next();
         if (kv != null && kv.getKey().startsWith(seekPrefix)) {
-
-          // Entry should not be marked for delete, consider only those
-          // entries.
-          CacheValue<OmKeyInfo> cacheValue =
-              keyTable.getCacheValue(new CacheKey<>(kv.getKey()));
-          if (cacheValue == null || cacheValue.getCacheValue() != null) {
+          if (cacheKeyMap.containsKey(kv.getKey())) {
```

Comment:
I don't get why  changes are needed here in listKeys ? Why does changing internal logic of PartialTableCache affect this ?